### PR TITLE
PEP8 and Coding Style Changes

### PR DESCRIPTION
**Pull request by cod3monk**
Issues found with PyCharm Inspection and partial manual review.

### CODE REVIEW

File: osaca/osaca.py [hunk 1]
```diff
@@ -484,23 +473,23 @@ def iaca_asm(self):
         code = self.srcCode
```
Comment (cod3monk) [inline osaca/osaca.py]: `srcCode` is initialized in `store_src_code()` as a List of Strings, but used here as a String. Is this intended?

File: osaca/osaca.py [hunk 2]
```diff
@@ -65,71 +65,70 @@ def include_ibench(self):
...
+        # Create sequence of numbers and their reciprocals for validate the measurements
         cyc_list, reci_list = self.create_sequences()
         print('Everything seems fine! Let\'s start!', file=self.file_output)
         new_data = []
         added_vals = 0
         for line in self.srcCode:
-            if('Using frequency' in line or len(line) == 0):
+            if 'Using frequency' in line or len(line) == 0:
                 continue
             clmn = 'LT'
```
Comment (cod3monk) [inline osaca/osaca.py]: column? Please use (mostly) complete words, spaced with underscores.

**Comment by cod3monk** [comment]
Variable `groups` is references but never initialized in `get_occurance_groups()` in `eu_sched.py`.

File: osaca/osaca.py (hunk 1 above)
Comment (JanLJL) [reply osaca/osaca.py]: `store_src_code()` stores `srcCode` is a String or a list of Strings depending on the given parameter `iacaFlag`, because in case of a marked kernel with IACA markers instead of high level OSACA markers, the regex is applied on `srcCode` as one string. Therefore this behavior is intended.

File: osaca/osaca.py (hunk 2 above)
Comment (JanLJL) [reply osaca/osaca.py]: changed `clmn` to `column`

File: osaca/osaca.py [hunk 3]
```diff
@@ -833,37 +819,35 @@ def main():
                        help='search for IACA markers instead the OSACA marker')
     group.add_argument('-m', '--insert-marker', dest='insert_marker', action='store_true',
                        help='try to find blocks probably corresponding to loops in assembly and'
-                       + 'insert IACA marker')
+                            + 'insert IACA marker')
     parser.add_argument('filepath', type=str, help='path to object (Binary, ASM, CSV)')
 
     # Store args in global variables
     inp = parser.parse_args()
-    if(inp.arch is None and inp.insert_marker is None):
```
Comment (JanLJL) [inline osaca/osaca.py]: In case of starting OSACA with `-m`  flag (for inserting IACA markers with kerncraft), no architecture is necessary. Therefore the check is valid. Added with the next commit.

File: osaca/osaca.py [hunk 4]
```diff
@@ -305,12 +301,8 @@ def write_csv(self, csv):
...
             CSV data as string
         """
-        #curr_dir = '/'.join(os.path.realpath(__file__).split('/')[:-1])
-        try:
-            f = open(self.osaca_dir+'data/'+self.arch.lower()+'_data.csv', 'w')
-        except IOError:
-            print('IOError: file \'{}\' not found in $HOME/.osaca/data'.format(self.arch.lower()
-                  + '_data.csv'), file=self.file_output)
+        # curr_dir = '/'.join(os.path.realpath(__file__).split('/')[:-1])
+        f = open(self.osaca_dir + 'data/' + self.arch.lower() + '_data.csv', 'w')
```
Comment (JanLJL) [inline osaca/osaca.py]: @cod3monk Why did you remove the error message here? I think it is helpful in case of something went wrong...

**Comment by JanLJL** [comment]
> Variable `groups` is references but never initialized in `get_occurance_groups()` in `eu_sched.py`.

`groups` is actually initialized in the master branch, it was deleted during commit bcdf673b9bfc9f58a995c9f2918aef1103f72b1b (_WIP: better pep8 compatability and coding style_), will be added with the next commit.

File: osaca/osaca.py (hunk 4 above)
Comment (cod3monk) [reply osaca/osaca.py]: Because the catch block only prints an error message and then does nothing to resolve the issue. The code execution will therefore be continued and will fail (with another exception) on `f.write(csv)`, which is misleading because it does not point at the actual problem. By not handling the exception at all, execution will be stopped and the user is presented with a very similar error message to yours (plus a stack trace for easy debugging, if needed).
If a `sys.exit(1)` would have been added to the catch block, it would have been fine with me.
